component/pipeline.go

```diff
@@ -0,0 +1,85 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package component // import "go.opentelemetry.io/collector/component"
```

Comment:
The concept of a pipeline seems a bit out of place in the component package. Have we considered what it would look like to create a new `pipeline` package?

Reply:
I totally agree - and I want to move this out.  Like I mentioned above:
> Followup Work: I wanted to move PipelineID to service/pipelines - but that would create a circular dependency with InstanceID, which uses PipelineID but lives in component.

I think service/pipelines should contain this and should depend on component, but I just didn't want to make those changes in this PR because its already long enough.

Reply:
Unfortunately if we put this in `component` and it gets released before the followup PR we'll have breaking changes when we remove it.

Since this PR already updates service to use this concept of pipeline, let's add PipelineID to service now.